src/runtime_src/core/pcie/driver/linux/xocl/subdev/xmc.c

```diff
@@ -2439,28 +2439,52 @@ static int load_xmc(struct xocl_xmc *xmc)
 
 	/* Wait for XMC to start
 	 * Note that ERT will start long before XMC so we don't check anything
+	 *
+	 * If dev tree has CMC_MUTEX register defined, we rely on the
+	 * regmap_ready bit to check whether cmc is ready, otherwise,
+	 * we still use the legacy 'init done' bit in REGMAP
 	 */
-	reg_val = READ_REG32(xmc, XMC_STATUS_REG);
-	if (!(reg_val & STATUS_MASK_INIT_DONE)) {
-		xocl_info(&xmc->pdev->dev, "Waiting for XMC to finish init...");
+	ret = xocl_iores_read32(xdev_hdl, XOCL_SUBDEV_LEVEL_BLD,
+		IORES_CMC_MUTEX, XOCL_RES_OFFSET_CHANNEL2, &reg_map_ready);
+	if (!ret) {
 		retry = 0;
-		while (retry++ < MAX_XMC_RETRY &&
-			!(READ_REG32(xmc, XMC_STATUS_REG) &
-			STATUS_MASK_INIT_DONE))
+		while (!ret && retry++ < MAX_XMC_RETRY &&
+			!(reg_map_ready & 0x2)) {
```

Comment:
given there are more flags instead of just 0x1, could we make the 0x2 more readable also the cmc_access_ops should have grant/ack defined also.

Reply:
talked to Brian, we've decided use another PR to clean those 0x1, 0x2 more readable.